flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/ResolvedAggLocalReference.java

```diff
@@ -20,19 +20,23 @@
 
 import org.apache.flink.table.expressions.Expression;
 import org.apache.flink.table.expressions.ExpressionVisitor;
+import org.apache.flink.table.expressions.ResolvedExpression;
+import org.apache.flink.table.types.DataType;
 import org.apache.flink.table.types.logical.LogicalType;
 
 import java.util.Collections;
 import java.util.List;
 
+import static org.apache.flink.table.runtime.types.LogicalTypeDataTypeConverter.fromLogicalTypeToDataType;
+
 /**
  * Special reference which represent a local filed, such as aggregate buffers or constants.
  * We are stored as class members, so the field can be referenced directly.
  * We should use an unique name to locate the field.
  *
  * <p>See {@link org.apache.flink.table.planner.codegen.ExprCodeGenerator#visitLocalRef}.
  */
-public class ResolvedAggLocalReference implements Expression {
+public class ResolvedAggLocalReference implements ResolvedExpression {
```

Comment:
I would like to keep the number of expression low. I'm wondering why this is needed. Can't we also use `FieldReferenceExpression` for these kinds of references? Having `nullTerm` and `fieldTerm` sounds very internal to me. Also for `ResolvedDistinctKeyReference`, why not expressing this like `'field.distinct`? I haven't looked deep into the code, people that are more familiar with the code base should decide.

Reply:
I mostly agree with @twalthr. I think though rather than `FieldReferenceExpression` we should use `LocalReferenceExpression`.

Reply:
`LocalReferenceExpression` is never be null because it is used for a group window in window aggregation.
But if it is used for aggregate buffer, it should be nullable, so we need a `nullTerm`. If not, how do we identify its null attribute?